libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/AlternativeRouteProgressDataProvider.kt

```diff
@@ -0,0 +1,37 @@
+package com.mapbox.navigation.core.routealternatives
+
+import com.mapbox.navigation.core.RouteProgressData
+
+internal object AlternativeRouteProgressDataProvider {
+
+    fun getRouteProgressData(
+        primaryRouteProgressData: RouteProgressData,
+        alternativeMetadata: AlternativeRouteMetadata
+    ): RouteProgressData {
+        val legIndex: Int
+        val routeGeometryIndex: Int
+        val legGeometryIndex: Int?
+        val primaryFork = alternativeMetadata.forkIntersectionOfPrimaryRoute
+        val alternativeFork = alternativeMetadata.forkIntersectionOfAlternativeRoute
+        val isForkPointAheadOnPrimaryRoute =
+            primaryRouteProgressData.routeGeometryIndex < primaryFork.geometryIndexInRoute
+        if (isForkPointAheadOnPrimaryRoute) {
+            val legIndexDiff = primaryFork.legIndex - alternativeFork.legIndex
+            legIndex = primaryRouteProgressData.legIndex - legIndexDiff
+            val routeGeometryIndexDiff =
+                primaryFork.geometryIndexInRoute - alternativeFork.geometryIndexInRoute
+            routeGeometryIndex =
+                primaryRouteProgressData.routeGeometryIndex - routeGeometryIndexDiff
```

Comment:
```suggestion
            val distanceToFork = primaryFork.geometryIndexInRoute - primaryRouteProgressData.routeGeometryIndex
            routeGeometryIndex = alternativeFork.geometryIndexInRoute - distanceToFork
            legGeometryIndex = alternativeFork.geometryIndexInLeg - distanceToFork
```
this would be much more readable to me